src/hooks/use-explorer.ts

```diff
@@ -101,20 +102,23 @@ type ExplorerState = {
 	setEsquerySelector: (esquerySelector: EsquerySelector) => void;
 };
 
+const getHashParams = (): URLSearchParams => {
+	return new URLSearchParams(location.hash.slice(1));
+};
+
 const hashStorage: StateStorage = {
 	getItem: (key): string => {
-		const searchParams = new URLSearchParams(location.hash.slice(1));
-		const storedValue = searchParams.get(key) ?? "";
+		const storedValue = getHashParams().get(key) ?? "";
 		return storedValue ? JSON.parse(atob(storedValue)) : "";
 	},
 	setItem: (key, newValue): void => {
-		const searchParams = new URLSearchParams(location.hash.slice(1));
+		const searchParams = getHashParams();
 		const encodedValue = btoa(JSON.stringify(newValue));
 		searchParams.set(key, encodedValue);
 		location.hash = searchParams.toString();
 	},
 	removeItem: (key): void => {
-		const searchParams = new URLSearchParams(location.hash.slice(1));
+		const searchParams = getHashParams();
```

Comment:
This looks like we're creating a new `URLSearchParams` for each method. Is that necessary or can the same one be reused?

Reply:
Yes, it is necessary to create a new URLSearchParams. If we reused a single URLSearchParams instance, it would contain stale data that doesn't reflect the current state of location.hash.